neuralforecast/tsdataset.py

```diff
@@ -371,12 +394,169 @@ def __init__(
         self.target_col = target_col
         self.min_size = min_size
 
-# %% ../nbs/tsdataset.ipynb 11
+# %% ../nbs/tsdataset.ipynb 10
+class IterativeTimeSeriesDataset(BaseTimeSeriesDataset):
+
+    def __init__(
+        self,
+        files_ds: _FilesDataset,
+        temporal_cols,
+        last_times,
+        indices,
+        max_size: int,
+        min_size: int,
+        y_idx: int,
+        static=None,
+        static_cols=None,
+        sorted=False,
+    ):
+        super().__init__(
+            temporal_cols=temporal_cols,
+            max_size=max_size,
+            min_size=min_size,
+            y_idx=y_idx,
+            static=static,
+            static_cols=static_cols,
+            sorted=sorted,
+        )
+        self.files_ds = files_ds
+        # array with the last time for each timeseries
+        self.last_times = last_times
+        self.indices = indices
+        self.n_groups = len(files_ds.files)
+
+    def __getitem__(self, idx):
+        if isinstance(idx, int):
+            df = pd.read_parquet(self.files_ds.files[idx])
+
+            _, _, data, _, _ = ufp.process_df(
+                df,
+                self.files_ds.id_col,
+                self.files_ds.time_col,
+                self.files_ds.target_col,
+            )
+
+            temporal_cols = self.files_ds.temporal_cols
+            # Add Available mask efficiently (without adding column to df)
+            if "available_mask" not in df.columns:
+                available_mask = np.ones((len(data), 1), dtype=np.float32)
+                data = np.append(data, available_mask, axis=1)
+                temporal_cols = temporal_cols.append(pd.Index(["available_mask"]))
+
+            data = self._as_torch_copy(data)
+
+            # Pad the temporal data to the left
+            temporal = torch.zeros(
+                size=(len(temporal_cols), self.files_ds.max_size), dtype=torch.float32
+            )
+            temporal[: len(temporal_cols), -len(data) :] = data.permute(1, 0)
+
+            # Add static data if available
+            static = None if self.static is None else self.static[idx, :]
+
+            item = dict(
+                temporal=temporal,
+                temporal_cols=temporal_cols,
+                static=static,
+                static_cols=self.static_cols,
+                y_idx=self.y_idx,
+            )
+
+            return item
+        raise ValueError(f"idx must be int, got {type(idx)}")
+
+    @staticmethod
+    def from_data_directory(
+        files,
+        static_df=None,
+        sort_df=False,
+        id_col="unique_id",
+        time_col="ds",
+        target_col="y",
+    ):
+        """We expect files_ds to have one parquet file per timeseries containing a pd df and static df to be a pd df"""
```

Comment:
Should probably use full names rather than acronyms in docstrings, ie "Pandas DataFrame" rather than pd df, makes things more readable, less translation required by the reader